backend/canisters/user_index/impl/src/lib.rs

```diff
@@ -137,18 +136,18 @@ struct Data {
     pub users: UserMap,
     pub service_principals: HashSet<Principal>,
     pub user_canister_wasm: CanisterWasm,
+    pub local_user_index_canister_wasm: CanisterWasm,
     pub sms_service_principals: HashSet<Principal>,
     pub sms_messages: EventStream<ConfirmationCodeSms>,
     pub group_index_canister_id: CanisterId,
     pub notifications_canister_ids: Vec<CanisterId>,
     pub canisters_requiring_upgrade: CanistersRequiringUpgrade,
-    pub canister_pool: canister::Pool,
     pub total_cycles_spent_on_canisters: Cycles,
     pub online_users_canister_ids: HashSet<CanisterId>,
     pub cycles_dispenser_canister_id: CanisterId,
     pub open_storage_index_canister_id: CanisterId,
     pub open_storage_user_sync_queue: OpenStorageUserSyncQueue,
-    pub user_event_sync_queue: UserEventSyncQueue,
+    pub user_index_event_sync_queue: CanisterEventSyncQueue<UserIndexEvent>,
```

Comment:
Same here

Reply:
We should deserialize from the old field to the new one so that events aren't lost